apps/studio/src/server/trpc.ts

```diff
@@ -96,6 +96,19 @@ const loggerWithVersionMiddleware = loggerMiddleware.unstable_pipe(
   },
 )
 
+const contentTypeHeaderMiddleware = t.middleware(async ({ ctx, next }) => {
+  if (
+    ctx.req.body &&
+    !ctx.req.headers['content-type']?.startsWith('application/json')
```

Comment:
if this says `application/jsononsense` i'll gladly accept it though?

Reply:
also this is a very narrow `content-type` to accept - how did we make this determination that we'd only accept this?

Reply:
This is based on the VAPT finding on KS! For any requests with a body, we always expect a JSON format - this is also the solution provided upstream by starter-kit (as well as for trpc 11)

Reply:
trpc only works with json, so we can't do other things like form-encoded etc.

Reply:
@alexanderleegs But to that earlier point should it be a strict equality check for application/json or a prefix check?

Reply:
Hmm i think either works since there's no other official MIME type beginning with application/json, but i've modified to be a strict equality check in https://github.com/opengovsg/isomer/pull/254/commits/f40692df65e569e5ec912a68dc78821fce8df7b4

Reply:
lgtm